Third party package updates release 1 14

vyaghras opened the pull request:
**Issue number:**

Closes: #2994

**Description of changes:**
```
packages:update containerd to 1.6.20
packages: update coreutils to 9.3
packages:update docker-proxy to f7cdd0e7adbf0818e384178093aa67e38ffdb112
packages: update ecr-credential-provider to 1.27.0
packages: update ecs-agent to 1.70.2
packages: update grep to 3.9
packages: update iproute2 to 6.2.0
packages: update libelf to 0.189
packages: update libglib to 2.76.0
packages: update libnftnl to 1.2.5
packages: update liblzma to 5.4.2
packages: update libcap to 2.68
packages:update libzstd to 1.5.5
packages:update libnvidia-container to 1.13.1
packages: update makedumpfile to 1.7.3
packages:update nvidia-k8s-device-plugin to 0.14.0
packages: update nvidia-container-toolkit to 1.13.0
packages: update open-vm-tools to 12.2.0
packages: update tcpdump to 4.99.4
packages: update strace to 6.2
```
Patch in coreutils have been deleted, as the change is available in upstream package. 

Updates all third party packages except for the following:

- binutils 2.38 -> 2.40 - bottlerocket-sdk has to move to 2.40 first.
- containerd 1.6.19 -> 1.7.0 - unstable version (https://github.com/containerd/containerd/releases)
- docker related packages(docker-cli, docker-engine) 20.12.21 -> 20.12.23 - To sync with ECS-optimized AMI packaged docker version listed in [its version table](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-ami-versions.html#ecs-ami-versions-linux).
- libncurses 6.2 -> 6.4 - See Update ncurses to 6.4 #1813
- libpcap 1.10.1 -> 1.10.3 - See update libpcap to 1.10.3 for *-dev variants #2827
- wicked 0.6.68 -> 0.6.70 - See update wicked to 0.6.70+ #2591
-  runc 1.1.4 -> 1.1.6 - See [#3037](https://github.com/bottlerocket-os/bottlerocket/pull/3037)

**Testing done:**
- [x] aws-k8s-1.24 x86_64 joins cluster.
- [x] aws-k8s-1.26 x86_64 joins cluster.
- [x] aws-k8s-1.24-nvidia aarch64 pass the quick tests.
```
NAME                                TYPE      STATE            PASSED         SKIPPED         FAILED    
 aarch64-aws-k8s-124-nvidia         Resource  completed                          
 aarch64-aws-k8s-124-nvidia-quick   Test      pass              1             6972             0      
```
- [x] aws-ecs-1  x86-64 pass the quick tests.
```
NAME                              TYPE           STATE      PASSED	SKIPPED    FAILED	 
x86-64-aws-ecs-1                  Resource       completed                        
x86-64-aws-ecs-1-instances-rtfq   Resource       completed                        
x86-64-aws-ecs-1-quick            Test           pass          1         0          0      
```
- [x] aws-ecs-1-nvidia x86-64 pass the quick tests.
```
NAME                                      TYPE          STATE         PASSED	SKIPPED    FAILED	  
x86-64-aws-ecs-1-nvidia                   Resource      completed                        
x86-64-aws-ecs-1-nvidia-instances-xafw    Resource      completed                         
x86-64-aws-ecs-1-nvidia-quick             Test          pass            1         0          0      
```
- [x] vmware-k8s-1.24, created cluster with eks-a with OVA built from changes and can run workloads fine
```
$ kubectl --kubeconfig br-eksa-124-eks-a-cluster.kubeconfig get nodes -o wide
NAME            STATUS   ROLES           AGE     VERSION                INTERNAL-IP     EXTERNAL-IP     OS-IMAGE                                   KERNEL-VERSION   CONTAINER-RUNTIME
198.19.132.87   Ready    <none>          62s     v1.24.10-eks-08ad9cc   198.19.132.87   198.19.132.87   Bottlerocket OS 1.13.0 (vmware-k8s-1.24)   5.15.90          containerd://1.6.19+bottlerocket
198.19.132.88   Ready    <none>          60s     v1.24.10-eks-08ad9cc   198.19.132.88   198.19.132.88   Bottlerocket OS 1.13.0 (vmware-k8s-1.24)   5.15.90          containerd://1.6.19+bottlerocket
198.19.33.21    Ready    control-plane   2m23s   v1.24.10-eks-08ad9cc   198.19.33.21    198.19.33.21    Bottlerocket OS 1.13.0 (vmware-k8s-1.24)   5.15.90          containerd://1.6.19+bottlerocket
198.19.66.43    Ready    control-plane   68s     v1.24.10-eks-08ad9cc   198.19.66.43    198.19.66.43    Bottlerocket OS 1.13.0 (vmware-k8s-1.24)   5.15.90          containerd://1.6.19+bottlerocket
```
- [x] vmware-k8s-1.26, created cluster with eks-a with OVA built from changes and all the sonobuoy tests passes.
```
15:35:38             e2e          global   complete            Passed:360, Failed:  0
```
- [x] metal-k8s, created cluster with eks-a with image built from changes and can run workloads fine
```
kubectl --kubeconfig br-125-3pup-eks-a-cluster.kubeconfig get nodes -o wide
NAME          STATUS   ROLES           AGE     VERSION               INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                  KERNEL-VERSION   CONTAINER-RUNTIME
10.80.50.22   Ready    <none>          2m40s   v1.25.6-eks-232056e   10.80.50.22   <none>        Bottlerocket OS 1.13.0 (metal-k8s-1.25)   5.15.90          containerd://1.6.19+bottlerocket
10.80.50.24   Ready    control-plane   9m43s   v1.25.6-eks-232056e   10.80.50.24   <none>        Bottlerocket OS 1.13.0 (metal-k8s-1.25)   5.15.90          containerd://1.6.19+bottlerocket
10.80.50.27   Ready    <none>          3m6s    v1.25.6-eks-232056e   10.80.50.27   <none>        Bottlerocket OS 1.13.0 (metal-k8s-1.25)   5.15.90          containerd://1.6.19+bottlerocket
```
**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

vyaghras committed:
packages: update grep to 3.9

vyaghras committed:
packages: update libelf to 0.189

vyaghras committed:
packages: update libglib to 2.76.0

vyaghras committed:
packages: update libnftnl to 1.2.5

vyaghras committed:
packages: update open-vm-tools to 12.2.0

vyaghras committed:
packages: update liblzma to 5.4.2

vyaghras committed:
packages:update docker-proxy to f7cdd0e7adbf0818e384178093aa67e38ffdb112

vyaghras committed:
packages: update iproute2 to 6.2.0

stmcginnis commented:
Looks like `strace` needs to be added to the lookaside cache? I think it's [only pulled in to the `*-dev` variants](https://github.com/bottlerocket-os/bottlerocket/blob/develop/variants/aws-dev/Cargo.toml#L36), so those are the only ones where it will fail to find it.

gazal-k commented:
> Updates all third party packages except for the following:
> - containerd 1.6.19 -> 1.7.0 - unstable version (https://github.com/containerd/containerd/releases)

Can we please update containerd to [1.6.20](https://github.com/containerd/containerd/releases/tag/v1.6.20)?

That would bump up `go-restful` to v3 and avoid the need for changes in #2813

vyaghras commented:
> > Updates all third party packages except for the following:
> > 
> > * containerd 1.6.19 -> 1.7.0 - unstable version (https://github.com/containerd/containerd/releases)
> 
> Can we please update containerd to [1.6.20](https://github.com/containerd/containerd/releases/tag/v1.6.20)?
> 
> That would bump up `go-restful` to v3 and avoid the need for changes in #2813

Thanks for bringing it up @gazal-k . We will check and get back on this.

jpmcb commented:
> That would bump up go-restful to v3 and avoid the need for changes 

I think upgrading to v1.16.20 of containerd is a good stopgap until we can make the leap to v1.7.0 but upgrading the 3rd party package won't affect the _code_ dependencies in the host-ctr go module. That will require a separate PR to host-ctr to upgrade there (and remove the `replace` for go-restful)

vyaghras committed:
packages:update containerd to 1.6.20

stmcginnis commented on packages/makedumpfile/makedumpfile.spec:
```diff
@@ -1,5 +1,5 @@
 Name: %{_cross_os}makedumpfile
-Version: 1.7.2
+Version: 1.7.3
```
Added to the lookaside cache, but looks like the new version includes more files that need to be handled in the `%files` section.

```
  #18 7.714 error: Installed (but unpackaged) file(s) found:
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/README
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/ap_messages_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/dir_names_3_10_to_3_13.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/dir_names_3_14_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/keyring_3_10_to_4_3.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/keyring_4_4_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/proc_names_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/tcp_sk_buf_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/udp_sk_buf_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/unix_sk_buff_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/vhost_net_buffers_3_10_to_3_18.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/vhost_net_buffers_3_19_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/eppic_scripts/vhost_scsi_buffers_3_10_to_4_8.c
  #18 7.714    /x86_64-bottlerocket-linux-gnu/sys-root/usr/share/makedumpfile/makedumpfile.conf.sample
```

stmcginnis replied:
I think we just need to ignore these two entries: https://github.com/makedumpfile/makedumpfile/blob/ecc9355e1344d3de94870e855e380ee84224e552/makedumpfile.spec#L41-L42

vyaghras replied:
Fixed this by changing spec file 👍

vyaghras committed:
packages: update makedumpfile to 1.7.3

vyaghras committed:
packages: update coreutils to 9.3

Also deleted the patch
1001-gnulib-simple-backup-fix.patch as it is available in new version of
upstream package

vyaghras committed:
packages: update libcap to 2.68

vyaghras committed:
packages:update libzstd to 1.5.5

vyaghras committed:
packages:update nvidia-k8s-device-plugin to 0.14.0

vyaghras committed:
packages: update ecr-credential-provider to 1.27.0

vyaghras committed:
packages: update ecs-agent to 1.70.2

vyaghras committed:
packages: update tcpdump to 4.99.4

vyaghras committed:
packages:update libnvidia-container to 1.13.1

vyaghras committed:
packages: update nvidia-container-toolkit to 1.13.0

vyaghras committed:
packages: update strace to 6.2

vyaghras commented:
> > Updates all third party packages except for the following:
> > 
> > * containerd 1.6.19 -> 1.7.0 - unstable version (https://github.com/containerd/containerd/releases)
> 
> Can we please update containerd to [1.6.20](https://github.com/containerd/containerd/releases/tag/v1.6.20)?
 @gazal-k We have updated containerd to 1.7.0 in this change.

stmcginnis reviewed (approved):
Looks good to me - thanks @vyaghras!

bcressey commented on packages/docker-proxy/docker-proxy.spec:
```diff
@@ -3,7 +3,7 @@
 %global goimport %{goproject}/%{gorepo}
 # Use the libnetwork commit listed in this file for the docker version we ship:
 # https://github.com/moby/moby/blob/DOCKER-VERSION-HERE/vendor.conf
-%global commit 0dde5c895075df6e3630e76f750a447cf63f4789
+%global commit f7cdd0e7adbf0818e384178093aa67e38ffdb112
```
Per the comment above this commit line, this needs to align with the Docker version, currently 20.10.21:
https://github.com/moby/moby/blob/v20.10.21/vendor.conf
